File: 169-majority-element/169-majority-element.ts
function majorityElement(nums: number[]): number {
    let max = nums[0];
    let maxCount = 1;
    for(let i = 0; i < nums.length; i++) {
        if(max === nums[i]) {
            maxCount++;
            continue;
        }
        maxCount--;
        if(maxCount === 0) {
           maxCount = 1;
            max = nums[i];
        }
    }
    return max;
};

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        max = nums[0]
        maxcount = 0
        
        for i in range(len(nums)):
            if max == nums[i]:
                maxcount += 1
                continue
            maxcount -= 1
            if maxcount == 0:
                maxcount = 1
                max = nums[i]
        return max
        